textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/nlpextrator/NlpExtractor.java

```diff
@@ -93,64 +132,108 @@ public ITuple getNextTuple() throws Exception {
+     * <p>
      * The NLP package has annotations for the start and end position of a token
      * and it perfectly matches the span design so we just use them.
-     *
+     * <p>
+     * For Example: With TextField value: "Microsoft, Google and Facebook are
+     * organizations while Donald Trump and Barack Obama are persons", with
+     * fieldName: Sentence1 and inputConstant is Organization. The flag would
+     * set to "NE" by constructor.
+     * The pipeline would set up to cover the Named Entity Recognizer. Then
+     * get the value of NamedEntityTagAnnotation for each CoreLabel(token).If
+     * the value is the constant "Organization", then it meets the
+     * requirement. In this cases "Microsoft","Google" and "Facebook" will
+     * satisfied the requirement. "Donald Trump" and "Barack Obama" would
+     * have constant "Person" and not meet the requirement. For each
+     * qualified token, create a span accordingly and add it to the return
+     * list. In this case, token "Microsoft" would have span:
+     * ["Sentence1", 0, 9, Organization, "Microsoft"]
      */
-    private List<Span> extractNESpans(IField iField, String fieldName) {
+    private List<Span> extractNlpSpans(IField iField, String fieldName) {
         List<Span> spanList = new ArrayList<>();
         String text = (String) iField.getValue();
         Properties props = new Properties();
-        props.setProperty("annotators", "tokenize, ssplit, pos, lemma, ner");
+
+        if (flag.equals("POS")) {
+            props.setProperty("annotators", "tokenize, ssplit, pos");
+        } else {
+            props.setProperty("annotators", "tokenize, ssplit, pos, lemma, " +
+                    "ner");
+        }
         StanfordCoreNLP pipeline = new StanfordCoreNLP(props);
         Annotation documentAnnotation = new Annotation(text);
         pipeline.annotate(documentAnnotation);
-        List<CoreMap> sentences = documentAnnotation.get(CoreAnnotations.SentencesAnnotation.class);
+        List<CoreMap> sentences = documentAnnotation.get(CoreAnnotations.
+                SentencesAnnotation.class);
         for (CoreMap sentence : sentences) {
-            for (CoreLabel token : sentence.get(CoreAnnotations.TokensAnnotation.class)) {
-                String NLPConstant = token.get(CoreAnnotations.NamedEntityTagAnnotation.class);
-                if (!NLPConstant.equals("O")) {
-
-                    String NEConstant = getNEConstant(NLPConstant);
-                    int start = token.get(CoreAnnotations.CharacterOffsetBeginAnnotation.class);
-                    int end = token.get(CoreAnnotations.CharacterOffsetEndAnnotation.class);
-                    String word = token.get(CoreAnnotations.TextAnnotation.class);
-
-                    Span span = new Span(fieldName, start, end, NEConstant, word);
-
-
-                    if (spanList.size() >= 1) {
+            for (CoreLabel token : sentence.get(CoreAnnotations
```

Comment:
Again, if there are formatting changes, try to avoid them in this PR (in general).